core/domain/state_domain.py

```diff
@@ -916,8 +940,917 @@ def get_all_outcomes(self) -> List[Outcome]:
+                            f'The rule \'{rule_spec_index}\' of answer '
+                            f'group \'{ans_group_index}\' having '
+                            f'rule type \'IsWithinTolerance\' '
+                            f'have \'tol\' value less than or equal to '
+                            f'zero in NumericInput interaction.'
+                        )
+                    self._set_lower_and_upper_bounds(
+                        range_var, rule_value_x - rule_value_tol,
+                        rule_value_x + rule_value_tol, True, True
+                    )
+
+                elif rule_spec.rule_type == 'IsInclusivelyBetween':
+                    rule_value_a = float(rule_spec.inputs['a'])
+                    rule_value_b = float(rule_spec.inputs['b'])
+                    # `a` should not be greater than equal to `b` in
+                    # `IsInclusivelyBetween` rule.
+                    if rule_value_a >= rule_value_b:
+                        raise utils.ValidationError(
+                            f'The rule \'{rule_spec_index}\' of answer '
+                            f'group \'{ans_group_index}\' having '
+                            f'rule type \'IsInclusivelyBetween\' '
+                            f'have `a` value greater than `b` value '
+                            f'in NumericInput interaction.'
+                        )
+                    self._set_lower_and_upper_bounds(
+                        range_var, rule_value_a,
+                        rule_value_b, True, True
+                    )
+
+                for range_ele in ranges:
+                    if self._is_enclosed_by(range_var, range_ele):
+                        raise utils.ValidationError(
+                            f'Rule \'{rule_spec_index}\' from answer '
+                            f'group \'{ans_group_index}\' will never be '
+                            f'matched because it is made redundant '
+                            f'by the above rules'
+                        )
+
+                ranges.append(range_var)
+
+    def _validate_fraction_input(self) -> None:
+        """Validates the FractionInput interaction.
+
+        Raises:
+            ValidationError. Duplicate rules are present.
+            ValidationError. Solution is not in simplest form when the
+                `simplest form` setting is turned on.
+            ValidationError. Solution is not in proper form when the
+                `proper form` setting is turned on.
+            ValidationError. `IsExactlyEqualTo` rule have integral value when
+                `allow non zero integers` setting is off.
+            ValidationError. Rule have solution that is subset of previous
+                rules' solutions
+            ValidationError. `HasFractionalPartExactlyEqualTo` rule comes after
+                `HasDenominatorEqualTo` rule where the fractional denominator is
+                equal to `HasDenominatorEqualTo` rule value.
```

Comment:
6 are mentioned here but in the function there are 7 errors

Reply:
Done